testutil/integration/router.go

```diff
@@ -0,0 +1,97 @@
+package integration
+
+import (
+	"fmt"
+
+	"github.com/cometbft/cometbft/abci/types"
+	cmtproto "github.com/cometbft/cometbft/proto/tendermint/types"
+
+	"cosmossdk.io/log"
+	storetypes "cosmossdk.io/store/types"
+	dbm "github.com/cosmos/cosmos-db"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+)
+
+type App struct {
+	*baseapp.BaseApp
+
+	ctx    sdk.Context
+	logger log.Logger
+
+	queryHelper *baseapp.QueryServiceTestHelper
+}
+
+func NewIntegrationApp(name string, logger log.Logger, keys map[string]*storetypes.KVStoreKey, modules ...module.AppModuleBasic) *App {
```

Comment:
This would be another good case for maybe two example tests showing how to call this using a standalone module, and another with a set of modules with dependencies.

While that will be partially covered in the package example, this would be a shorter and more direct example focused on calling `NewIntegrationApp`. In this case, it is probably less important to ensure the example is runnable, i.e. it's probably okay to not have the `// Output:` line at the end.